library/general/src/modules/PackagesProposal.rb

```diff
@@ -144,35 +142,20 @@ def CheckParams(unique_ID, type)
     #
     # @see #supported_resolvables
     # @see #RemoveResolvables()
-    def AddResolvables(unique_ID, type, resolvables)
+    def AddResolvables(unique_ID, type, resolvables, optional: false)
       resolvables = deep_copy(resolvables)
       return false if !CheckParams(unique_ID, type)
 
-      CreateEmptyStructureIfMissing(unique_ID, type)
+      CreateEmptyStructureIfMissing(unique_ID, type, optional: optional)
 
       if resolvables.nil?
-        Builtins.y2warning("Changing resolvables %1 to empty list", resolvables)
+        log.info("Using empty list instead of nil")
         resolvables = []
       end
 
-      Builtins.y2milestone(
-        "Adding resolvables %1 type %2 for %3",
-        resolvables,
-        type,
-        unique_ID
-      )
-      Ops.set(
-        @resolvables_to_install,
-        [unique_ID, type],
-        Convert.convert(
-          Builtins.union(
-            Ops.get(@resolvables_to_install, [unique_ID, type], []),
-            resolvables
-          ),
-          from: "list",
-          to:   "list <string>"
-        )
-      )
+      log.info("Adding #{log_label(optional)} #{resolvables} of type #{type} for #{unique_ID}")
+
+      data(optional)[unique_ID][type].concat(resolvables)
```

Comment:
it would be nice if `CreateEmptyStructureIfMissing(unique_ID, type, optional: optional)` just return that `data(optional)[unique_ID][type]` so maybe better name would be `data_for_uid_and_type` which create it if needed.